android/services/mongodb-remote-perftests/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/performance/SyncPerformanceIntTestsHarness.kt

```diff
@@ -249,26 +284,43 @@ class PartialResult {
     var networkReceived = 0.0
 }
 
-open class RunResult(numDocs: Int, docSize: Int, numOutliers: Int) {
+open class RunResult(numDocs: Int, docSize: Int) {
     val runTimes = arrayListOf<Double>()
     val networkSentBytes = arrayListOf<Double>()
     val networkReceivedBytes = arrayListOf<Double>()
     val memoryUsages = arrayListOf<Double>()
     val diskUsages = arrayListOf<Double>()
     var activeThreadCounts = arrayListOf<Double>()
+    var failures = arrayListOf<FailureResult>()
+    val numOutliers = SyncPerformanceTestUtils.getNumOutliers()
 
     val asBson by lazy {
-        Document(
-            mapOf(
-                "numDocs" to BsonInt32(numDocs),
-                "docSize" to BsonInt32(docSize),
-                "timeMs" to DataBlock(runTimes.toDoubleArray(), numOutliers).toBson(),
-                "networkSentBytes" to DataBlock(networkSentBytes.toDoubleArray(), numOutliers).toBson(),
-                "networkReceivedBytes" to DataBlock(networkReceivedBytes.toDoubleArray(), numOutliers).toBson(),
-                "memoryBytes" to DataBlock(memoryUsages.toDoubleArray(), numOutliers).toBson(),
-                "diskBytes" to DataBlock(diskUsages.toDoubleArray(), numOutliers).toBson(),
-                "activeThreadCounts" to DataBlock(activeThreadCounts.toDoubleArray(), numOutliers).toBson()
+        if (failures.size < (SyncPerformanceTestUtils.getNumIters() + 1) / 2) {
```

Comment:
I don't follow this logic, please explain?

Reply:
Just that we consider the test a failure (dont get its metrics) if it fails half of the iterations